02_cancer_type_classification/lasso_range_analysis/nbconverted/lasso_range_all.py

```diff
@@ -0,0 +1,280 @@
+
+# In[1]:
+
+
+import os
+import itertools as it
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from scipy.stats import pearsonr, spearmanr
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.data_utilities as du
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+base_results_dir = os.path.join(
+    cfg.repo_root, '02_cancer_type_classification', 'results', 'lasso_range_valid'
+)
+
+training_dataset = 'all_other_cancers'
+results_dir = os.path.join(base_results_dir, training_dataset)
+
+# cutoff to filter out "dummy regressor" over-regularized models
+# these can deflate performance around feature count 0, which can lead to
+# spurious positive correlations between model size and performance
+# set to None for no cutoff
+quantile_cutoff = 0.01
+
+# 'aupr' or 'auroc'
+metric = 'aupr'
+
+# 'pearson', 'spearman', or 'ccc'
+correlation = 'pearson'
+
+output_plots = True
+output_plots_dir = cfg.cancer_type_lasso_range_dir
+
+
+# ### Get coefficient information for each lasso penalty
+
+# In[3]:
+
+
+nz_coefs_df = []
+
+# get pancancer coefs info for now
+for coef_info in au.generate_nonzero_coefficients_lasso_range(results_dir):
+    (gene,
+     cancer_type,
+     seed,
+     lasso_param,
+     coefs_list) = coef_info
+    for fold_no, coefs in enumerate(coefs_list):
+        nz_coefs_df.append(
+            [gene, cancer_type, lasso_param, seed, fold_no, len(coefs)]
+        )
+        
+nz_coefs_df = pd.DataFrame(
+    nz_coefs_df,
+    columns=['gene', 'cancer_type', 'lasso_param', 'seed', 'fold', 'nz_coefs']
+)
+print(nz_coefs_df.shape)
+print(nz_coefs_df.gene.unique())
+nz_coefs_df.head()
+
+
+# ### Get performance information for each lasso penalty
+
+# In[4]:
+
+
+perf_df = au.load_prediction_results_lasso_range(results_dir, training_dataset)
+
+print(perf_df.shape)
+print(perf_df.gene.unique())
+perf_df.head()
+
+
+# ### Compare feature selection with performance
+
+# In[5]:
+
+
+coefs_perf_df = (nz_coefs_df
+    .rename(columns={'cancer_type': 'holdout_cancer_type'})
+    .merge(perf_df[perf_df.signal == 'signal'],
+           on=['gene', 'holdout_cancer_type', 'seed', 'fold', 'lasso_param'])
+    .drop(columns=['signal', 'experiment'])
+)
+
+print(coefs_perf_df.shape)
+coefs_perf_df.head()
+
+
+# In[6]:
+
+
+sns.set({'figure.figsize': (8, 6)})
+
+sns.histplot(coefs_perf_df.nz_coefs)
+plt.title('Distribution of feature count across cancer types/folds')
+plt.xlabel('Number of nonzero features')
+
+# calculate quantile cutoff if included
+if quantile_cutoff is not None:
+    nz_coefs_cutoff = coefs_perf_df.nz_coefs.quantile(q=quantile_cutoff)
+    plt.gca().axvline(nz_coefs_cutoff, linestyle='--')
+    print('cutoff:', nz_coefs_cutoff)
+    
+coefs_perf_df.loc[coefs_perf_df.nz_coefs.sort_values()[:8].index, :]
```

Comment:
Based on the comment above, shouldn't the rows with 0 and 1 nz_coefs (I'm assuming means nonzero coefficients) be thrown out?

Reply:
Yeah, that happens in the next cell - here I wanted to visualize a few of the rows that are being thrown out, before they're actually thrown out.  I'll add a comment saying this.